addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -22,7 +22,8 @@ def default_get(self, field_list):
 
     @api.onchange('project_id')
     def onchange_project_id(self):
-        self.task_id = False
+        if self.project_id != self.task_id.project_id:
```

Comment:
`if self.task_id.project_id and self.project_id != self.task_id.project_id:`

Reply:
or perhaps:
<code>if (self.task_id.project_id and self.project_id != self.task_id.project_id) or not self.task_id.project_id:</code>